graphium/finetuning/finetuning.py

```diff
@@ -0,0 +1,56 @@
+from typing import Iterable, List, Dict, Tuple, Union, Callable, Any, Optional, Type
+
+import pytorch_lightning as pl
+
+from torch.optim.optimizer import Optimizer
+from pytorch_lightning.callbacks import BaseFinetuning
+
+
+class MolecularFinetuning(BaseFinetuning):
+    def __init__(
+        self,
+        cfg,
+        train_bn: bool = False
+
+    ):
+        """
+        Finetuning logic
+        """
+        super().__init__()
+
+        cfg_finetune = cfg['trainer']['finetuning']
+        self.depth = cfg_finetune['added_layers']['depth']
```

Comment:
Add some checks in the `__init__` that `depth` and `add_fine_depth` and `on_task` are consistent with the model. Perhaps this would require to pass the model at the `__init__`, which is weird, but we want the code to crash BEFORE we start processing the data and training the model. We want a quick crash, instead of waiting 20min+ before it crashes for a wrong depth.